pkg/apis/alicloud/v1alpha1/types_infrastructure.go

```diff
@@ -113,8 +113,10 @@ type NatGatewayConfig struct {
 type InfrastructureStatus struct {
 	metav1.TypeMeta `json:",inline"`
 
-	VPC         VPCStatus `json:"vpc"`
-	KeyPairName string    `json:"keyPairName"`
+	VPC VPCStatus `json:"vpc"`
+
+	// KeyPairName should not be used as ssh-key-pair will not generated by terraform.
```

Comment:
I used to set it as `Deprecated`. But I can't pass the `make check` because the generated files uses this field and I have no chance to mark it as `no-lint`.

Reply:
I see. Then let's mark the field as deprecated and use nolint in the few usages.

```diff
--- a/pkg/controller/infrastructure/actuator.go
+++ b/pkg/controller/infrastructure/actuator.go
@@ -303,8 +303,7 @@ func (a *actuator) ensureOldSSHKeyDetached(ctx context.Context, infra *extension
                return err
        }

-       // nolint
-       if infrastructureStatus.KeyPairName == "" {
+       if infrastructureStatus.KeyPairName == "" { //nolint:staticcheck
                return nil
        }

@@ -318,9 +317,9 @@ func (a *actuator) ensureOldSSHKeyDetached(ctx context.Context, infra *extension
                return err
        }

-       a.logger.V(2).Info("Detaching ssh key pair from ECS instances", "keypair", infrastructureStatus.KeyPairName)
-       err = shootAlicloudECSClient.DetachECSInstancesFromSshKey(infrastructureStatus.KeyPairName)
-       a.logger.V(2).Info("Finished detaching ssh key pair from ECS instances", "keypair", infrastructureStatus.KeyPairName)
+       a.logger.V(2).Info("Detaching ssh key pair from ECS instances", "keypair", infrastructureStatus.KeyPairName)          //nolint:staticcheck
+       err = shootAlicloudECSClient.DetachECSInstancesFromSshKey(infrastructureStatus.KeyPairName)                           //nolint:staticcheck
+       a.logger.V(2).Info("Finished detaching ssh key pair from ECS instances", "keypair", infrastructureStatus.KeyPairName) //nolint:staticcheck

        return err
 }
```

Reply:
I don't think this comment is resolved.

Reply:
I can't do it for `generated` files. I will remove it in future..

Reply:
> I can't do it for generated files. I will remove it in future..

Did you checked this locally?

Reply:
For me, `make check` is passing with the following diff:

```diff
diff --git a/pkg/apis/alicloud/types_infrastructure.go b/pkg/apis/alicloud/types_infrastructure.go
index ef83404b..e1e89fe3 100644
--- a/pkg/apis/alicloud/types_infrastructure.go
+++ b/pkg/apis/alicloud/types_infrastructure.go
@@ -112,7 +112,9 @@ type InfrastructureStatus struct {
 
 	VPC VPCStatus
 
-	// KeyPairName should not be used as ssh-key-pair will not generated by terraform.
+	// KeyPairName is the key pair name.
+	//
+	// Deprecated: the field should not be used anymore as the SSH key pair is no longer generated by terraform.
 	KeyPairName string
 
 	// MachineImages is a list of machine images that have been used in this infrastructure. Usually, the extension controller
diff --git a/pkg/apis/alicloud/v1alpha1/types_infrastructure.go b/pkg/apis/alicloud/v1alpha1/types_infrastructure.go
index 71920833..2a099ebc 100644
--- a/pkg/apis/alicloud/v1alpha1/types_infrastructure.go
+++ b/pkg/apis/alicloud/v1alpha1/types_infrastructure.go
@@ -115,7 +115,9 @@ type InfrastructureStatus struct {
 
 	VPC VPCStatus `json:"vpc"`
 
-	// KeyPairName should not be used as ssh-key-pair will not generated by terraform.
+	// KeyPairName is the key pair name.
+	//
+	// Deprecated: the field should not be used anymore as the SSH key pair is no longer generated by terraform.
 	KeyPairName string `json:"keyPairName"`
 
 	// MachineImages is a list of machine images that have been used in this infrastructure. Usually, the extension controller
diff --git a/pkg/controller/infrastructure/actuator.go b/pkg/controller/infrastructure/actuator.go
index 9ddc4543..38acbe66 100644
--- a/pkg/controller/infrastructure/actuator.go
+++ b/pkg/controller/infrastructure/actuator.go
@@ -303,7 +303,7 @@ func (a *actuator) ensureOldSSHKeyDetached(ctx context.Context, infra *extension
 		return err
 	}
 
-	if infrastructureStatus.KeyPairName == "" {
+	if infrastructureStatus.KeyPairName == "" { //nolint:staticcheck
 		return nil
 	}
 
@@ -317,9 +317,9 @@ func (a *actuator) ensureOldSSHKeyDetached(ctx context.Context, infra *extension
 		return err
 	}
 
-	a.logger.V(2).Info("Detaching ssh key pair from ECS instances", "keypair", infrastructureStatus.KeyPairName)
-	err = shootAlicloudECSClient.DetachECSInstancesFromSSHKeyPair(infrastructureStatus.KeyPairName)
-	a.logger.V(2).Info("Finished detaching ssh key pair from ECS instances", "keypair", infrastructureStatus.KeyPairName)
+	a.logger.V(2).Info("Detaching ssh key pair from ECS instances", "keypair", infrastructureStatus.KeyPairName)          //nolint:staticcheck
+	err = shootAlicloudECSClient.DetachECSInstancesFromSSHKeyPair(infrastructureStatus.KeyPairName)                       //nolint:staticcheck
+	a.logger.V(2).Info("Finished detaching ssh key pair from ECS instances", "keypair", infrastructureStatus.KeyPairName) //nolint:staticcheck
 
 	return err
 }
```